simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -196,7 +196,10 @@ template <class T>
 template <class T>
 [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION simd<T, simd_abi::scalar> abs(
     simd<T, simd_abi::scalar> const& a) {
-  return simd<T, simd_abi::scalar>(std::abs(static_cast<T>(a)));
+  if constexpr (std::is_signed_v<T>)
+    return simd<T, simd_abi::scalar>(std::abs(static_cast<T>(a)));
+  else
+    return simd<T, simd_abi::scalar>(static_cast<T>(a));
```

Comment:
`return a;`?

Reply:
Changed to manually compute abs() since this needs to be a device callable function.